src/bigunit.ts

```diff
@@ -399,27 +417,89 @@ export class BigUnit {
     // If the length is less than the precision, pad with zeros
     if (length < this.precision) {
       const padding = "0".repeat(this.precision - length);
-      return `0.${padding}${valueString}`;
+      // If the value is negative, add a negative sign to the padded zeros and remove the negative sign from the value
```

Comment:
Could we try to use some logic that checks to see if `this.value < 0` early on rather than waiting for the incorrect string to be generated and fixing?